core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksInodeStore.java

```diff
@@ -514,20 +566,24 @@ public CheckpointName getCheckpointName() {
 
   @Override
   public void writeToCheckpoint(OutputStream output) throws IOException, InterruptedException {
-    mRocksStore.writeToCheckpoint(output);
+    try (RocksExclusiveLockHandle lock = mRocksStore.lockForRestart()) {
```

Comment:
Writing to checkpoint will acquire an exclusive lock and therefore all ongoing operations will be aborted. It does sound a bit awkward if we phrase it this way.

However, I think this does not really happen, where "this" means taking a checkpoint will fail all ongoing iterations. First of all, the shared state lock will be acquired before taking the checkpoint. And by being able to take that lock, we've already made sure there's no RPC. Also, on standby masters where checkpoints are taken, there will not be ongoing RPCs.

Reply:
Yes there should be no operations happening during checkpointing or restoring from checkpointing, otherwise the checkpoint would be invalid anyway